### PR TITLE
update the 7-Train-traditional-ML-models-using-the-Merlin-Models-API notebook based on customer feedback in #906

### LINKED ISSUE #906: [DOC] Revise and improve 07-Train-traditional-ML-models-using-the-Merlin-Models-API notebook
Client feedback suggests that the portion of this notebook on training implicit models is rather limited. It currently includes minimal steps to "fit" and "train" two implicit models + LightFM with some parameters (eg. warp loss) set, but not explained in detail, and according to the client, the lack of detail makes it hard to follow what exactly is going on. Could we:

- add further text explaining more about the models we included in this section (and why), and provide links (for the two implicit models) for reference
- add text explaining why some model parameters are chosen as they are (with other possible options) and what some of the hidden/assumed parameters are
- show data records (eg. train.head) going into the training as well as results of prediction and brief explanations
- show evaluation metrics (not just the line to execute metric calculation) and interpret them
- describe limitations of the current implicit API in Merlin Models. For example, based on the way the wrapper code is currently written in Merlin Models, there is an assumption that we're in a user-item recommender setting and item-item recommender setting is missing. However, in Ben's library there is model.similar_items(itemid) functionality.

### PR DESCRIPTION
**Pull request by radekosmulski**
resolves #906

### CODE REVIEW

File: examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb [hunk 1]
```diff
@@ -2,7 +2,7 @@
  "cells": [
```
Comment (bbozkaya) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>Revise typo as: "There are two ways we can train an implicit model."</p><p>Revise typo as: "There are two implicit models you can train with Merlin Models:"</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/911/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (bbozkaya) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>Add the names of the metrics to the notebook text as well (right after the "Implicit models can be..." sentence). It would also help to give a link to the definitions of these metrics, no matter how commonly known we think they are.</p><p> </p><p>Also, do we want to say anything about the significance or strength of the values of these metrics. They seem a bit low to indicate strong prediction performance in a real-life setting, but for the purpose of a demo notebook, they seem ok.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/911/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (bbozkaya) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>Revise typo as: "let's"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/911/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (bbozkaya) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>Suggestion: output the head of valid before the predict call and describe what is being passed (user ids). The prediction output is a list of item ids, per each user, to be interacted with, ordered from most likely to least likely (indicated by the values in the second array).</p><p> </p><p>Also, what is the meaning or definition of the numbers 4.0, 3.9, 3.89, etc.? They are not probabilities, so what are they? Are they scaled out of 100, etc.? A bit more explanation here so the user can make a better sense of these numbers. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/911/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (bbozkaya) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>Are these (precisions@10, auc) the default evaluation metrics? What other metrics are available and how does one specify them? (A link to a page might suffice).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/911/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (bbozkaya) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>These are two of the same metrics used above, right? Shall we add a short note here to mention that they are both higher than their implicit counterparts, which means lightfm performed better? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/911/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (bbozkaya) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>Mention that we are again using the same validation data set with the same structure and provide an interpretation of the values -3.25, -2.33, etc... (which are clearly different from the output of <code>implicit</code>) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/911/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>corrected</p><p> </p>

Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>metrics listed and definitions linked</p>

Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>corrected</p>

Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>added text addressing the points you raise</p>

Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>these are the default metrics available in our implementation, no other metrics are available nor can this list of metrics be customized</p>

Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>I wouldn't be inclined to make such a statement in a notebook aimed at being a tutorial. You generally need to be very careful when evaluating the performance of different frameworks, make sure you tuned the hyperparameters properly, make sure the comparison is fair, etc. Plus there is no signal to the user here, IMO. The performance is likely to vary greatly between these two algorithms given the size of the dataset, relationships in the dataset, etc. Possibly it is best strategy to try both models, especially given how little effort that costs (a user can use our high level API with the same dataset).</p>

Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>explanation added</p>

[the remark this answers is not in the document]
Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>good point, removed the word "likelihood" alltogether not to suggest a probabilistic interpretation</p>

[the remark this answers is not in the document]
Comment (radekosmulski) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:2]: <p>thank you for the suggestion, added a bit more explanation there! 🙂</p>

Comment (bbozkaya) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb]: <p>There is one more typo to fix with the word "Merli", right before the 2 bullets.</p>

Comment (bbozkaya) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb]: <p>This looks great. The only minor correction I'd point out is with the word "acti" in the paragraphs that starts with "Essentially,...". I believe you meant to say "actions".</p>

Comment (bbozkaya) [reply examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb]: <p>Thanks for the explanations on ids, scores, etc.; it reads well. I noticed a double "the" in the second last paragraph, please remove one.</p><p> </p>

File: examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb [hunk 2]
```diff
@@ -438,43 +733,263 @@
...
+    "implicit_preds = implicit.predict(valid)\n",
+    "implicit_preds"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28cc48df",
+   "metadata": {},
+   "source": [
+    "The predictions are item ids for each of the user id that we passed in as our data. They are ordered from the most likely to be interacted with by the user to the least likely as predicted by our `implicit` model. These IDs are contained in the the first array.\n",
```
Comment (rnyak) [inline examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb:990]:
```suggestion
    "The predictions are item ids for each of the user id that we passed in as our data. They are ordered from the most likely to be interacted with by the user to the least likely as predicted by our `implicit` model. These IDs are contained in the first array.\n",
```

**Review by rnyak — APPROVED** [review]

**Review by bbozkaya — APPROVED** [review]

**Comment by radekosmulski** [comment]
rerun tests

**Comment by radekosmulski** [comment]
rerun tests

**Comment by radekosmulski** [comment]
rerun tests
